cider/include/cider/batch/ScalarBatch.h

```diff
@@ -85,4 +87,76 @@ class ScalarBatch final : public CiderBatch {
   }
 };
 
+template <>
+class ScalarBatch<CiderVarchar> final : public CiderBatch {
```

Comment:
Emm, why not derive a new class like ```VarCharBatch```, because ```VarCharBatch``` and ```ScalarBatch``` have different buffer layout and resize logic.